### PR TITLE
[minor][PS4-1495] feat: Configuration for staging and production

### PR DESCRIPTION
**Pull request by satyadeeproat**

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -18,26 +18,57 @@ jobs:
...
 
-  publish-release:
+  publish-staging-release:
     working_directory: ~/workspace
     docker:
       - image: cimg/base:stable
     steps:
       - checkout
       - gh/setup
+      - add_ssh_keys:
```
Comment (ggeorgiadis) [inline .circleci/config.yml:28]: is this needed? If so, don't we need an actual fingerprint?

File: publish-release.sh [hunk 2]
```diff
@@ -32,13 +35,17 @@ increment_version() {
...
+  prerelease_flag="--prerelease"
 fi
 
-release_url=$(gh release create "$new_version" ./providers/*.jar --title="$new_version" --notes="$commit_message" $cmd_args)
+echo "New version: $new_version"
+echo "Commit message: $commit_message"
+echo "Prerelease flag: $prerelease_flag"
 
-echo $release_url
+release_url=$(gh release create "$new_version" ./providers/*.jar --title="$new_version" --notes="$commit_message" $prerelease_flag)
```
Comment (ggeorgiadis) [inline publish-release.sh]: nit: If that fails like last time, could we check the output and if it's not successful, we return a non success code? That way circleci will notify us

File: .circleci/config.yml (hunk 1 above)
Comment (satyadeeproat) [reply .circleci/config.yml:28]: The recommended way in CircleCI was to have private key added in the project and public key in github repo then add this in steps.

Comment (ggeorgiadis) [reply .circleci/config.yml:28]: ah ok, then let's use a unique fingerprint for each environment

Comment (ggeorgiadis) [reply .circleci/config.yml:28]: unless ofc they use the same key

File: publish-release.sh (hunk 2 above)
Comment (satyadeeproat) [reply publish-release.sh]: Updated the PR

**Review by ggeorgiadis — APPROVED** [review]
nicely done 👏🏻

**Pull request merged**